src/main/java/com/internship/socialnetwork/dto/CommentDto.java

```diff
@@ -0,0 +1,49 @@
+package com.internship.socialnetwork.dto;
+
+import com.internship.socialnetwork.exception.BadRequestException;
+import com.internship.socialnetwork.model.Comment;
+import jakarta.validation.constraints.NotBlank;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+
+@Data
+@AllArgsConstructor
+@Builder
+public class CommentDto {
+
+    private Long id;
+
+    @NotBlank(message = "Comment text is empty!")
+    private String text;
+
+    @NotBlank
+    private String authorUsername;
+
+    private Long parentCommentId;
+
+    public static CommentDto convertToDto(Comment comment) {
+       CommentDto commentDto = CommentDto.builder()
+                .id(comment.getId())
+                .text(comment.getText())
+                .authorUsername(comment.getAuthor().getUsername())
+                .build();
+
+       if(comment.getParentComment() != null) {
```

Comment:
This can be an inline check with Optional.ofNullable.ALso when you write ifs, there needs to be space between if and open parenthesis